common/lib/xmodule/xmodule/static_content.py

```diff
@@ -124,6 +124,8 @@ def _write_js(output_root, classes):
         module_js = class_.get_javascript()
         for filetype in ('coffee', 'js'):
             for idx, fragment in enumerate(module_js.get(filetype, [])):
+                if filetype != 'js':
```

Comment:
This is not a legitimate fix. There will be other JS files in the future, as we are slowing converting coffee files to JS over time. We need to enforce that specifically xmodule.js comes before all the files with classes that extend XModuleDescriptor.

:-1: